src/components/graph/graph.coords.js

```diff
@@ -0,0 +1,333 @@
+ */
+function calcRectangleVectorLengthFromCoords({ x1, y1, x2, y2 }, { px, py }, directionVector) {
+  const angle = Math.atan(directionVector.y / directionVector.x);
+
+  const vx = Math.cos(angle);
+  const vy = Math.sin(angle);
+
+  if (vx === 0) {
+    return x2 - x1;
+  } else if (vy === 0) {
+    return y2 - y1;
+  }
+
+  const leftEdge = (x1 - px) / vx;
+  const rightEdge = (x2 - px) / vx;
+
+  const topEdge = (y1 - py) / vy;
+  const bottomEdge = (y2 - py) / vy;
+
+  return Math.min(...[leftEdge, rightEdge, topEdge, bottomEdge].filter(edge => edge > 0));
+}
+
+/**
+ * Calculate a the vector length from the center of a circle to it's perimeter.
+ *
+ * @param {number} nodeSize The size of the circle, when no viewGenerator is specified, else the size of an edge of the viewGenerator square.
+ * @param {boolean} isCustomNode is viewGenerator specified.
+ * @returns {number} The length of the vector from the center of the symbol to it's closet edge, considering the given direction vector.
+ */
+function calcCircleVectorLength(nodeSize, isCustomNode) {
+  let radiusLength;
+  if (isCustomNode) {
+    // nodeSize equals the Diameter in the case of custome-node.
+    radiusLength = nodeSize / 10 / 2;
+  } else {
+    // because this is a circle and A = pi * r^2
+    // we multiply by 0.95, because if we don't the link is not melting properly
+    radiusLength = Math.sqrt(nodeSize / Math.PI);
+  }
+  return radiusLength;
+}
+
+/**
+ * Calculate a the vector length from the center of a square to it's perimeter.
+ *
+ * @param {number} nodeSize The size of the square, when no viewGenerator is specified, else the size of an edge of the viewGenerator square.
+ * @param {Object.<string, number>} nodeCoords The coords of a the square node.
+ * @param {Object.<string, number>} directionVector a 2D vector with x and y components
+ * @param {boolean} isCustomNode is viewGenerator specified.
+ * @returns {number} The length of the vector from the center of the symbol to it's closet edge, considering the given direction vector.
+ */
+function calcSquareVectorLength(nodeSize, { x, y }, directionVector, isCustomNode) {
+  let edgeSize;
+  if (isCustomNode) {
+    // nodeSize equals the edgeSize in the case of custome-node.
+    edgeSize = nodeSize / 10;
+  } else {
+    // All the edges of a square are equal, inorder to calc it's size we multplie two edges.
+    edgeSize = Math.sqrt(nodeSize);
+  }
+
+  // The x and y coords in this library, represent the top center of the component.
+  const leftSquareX = x - edgeSize / 2;
+  const topSquareY = y - edgeSize / 2;
+
+  return calcRectangleVectorLengthFromCoords(
+    { x1: leftSquareX, y1: topSquareY, x2: leftSquareX + edgeSize, y2: topSquareY + edgeSize },
+    { px: x, py: y },
+    directionVector
+  );
+}
+
+/**
+ * Calculate a the vector length from the center of a rectangle to it's perimeter.
```

Comment:
Calculates the vector length from the center of a rectangle to the closest edge following a direction.